### PR TITLE
Use SparkSession instead of SparkContext

### PR DESCRIPTION
**Pull request by acmiyaguchi**
The default tests fail to run due to multiple spark contexts. This uses the consolidated SparkSession object from Spark 2.0 and creates a shared spark context.

### CODE REVIEW

File: tests/test_main.py [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
+
+
 @pytest.fixture(scope="session")
 def spark_context(request):
-    conf = SparkConf().setMaster("local")\
-        .setAppName("python_etl" + "_test")
-    sc = SparkContext(conf=conf)
+    spark = (SparkSession
+             .builder
+             .appName("churn_test")
```
Comment (harterrt) [inline tests/test_main.py]: Can you change this to "python_etl_test"?

Comment (acmiyaguchi) [inline tests/test_main.py]: Done.

**Comment by acmiyaguchi** [comment]
I think the issue occurs when there are tests running in parallel. I saw the issue when I added a new test file for a job that I'm porting over.

**Comment by harterrt** [comment]
Sorry for the delay. There was a bug in my environment that was causing these tests to fail. This looks great, thanks for tackling this!

**Pull request merged**